src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
+
+    CWalletDBWrapper& dbw = wallet->GetDBHandle();
+    CWalletDB(dbw).WriteNameFirstUpdate(strName, jsonData);
+
+    return tr("");
+}
+
+std::string WalletModel::completePendingNameFirstUpdate(std::string &name, std::string &rand, std::string &txid, std::string &data, std::string &toaddress)
+{
+    JSONRPCRequest jsonRequest;
+    UniValue params(UniValue::VOBJ);
+    UniValue res;
+    std::string errorStr;
+
+    params.push_back (Pair("name", name));
+    params.push_back (Pair("rand", rand));
+    params.push_back (Pair("tx", txid));
+    params.push_back (Pair("value", data));
+    if(!toaddress.empty())
+        params.push_back (Pair("toaddress", toaddress));
+
+    jsonRequest.strMethod = std::string("name_firstupdate");
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    LogPrintf("executing name_firstupdate name=%s rand=%s tx=%s value=%s\n",
+        name.c_str(), rand.c_str(), txid.c_str(), data.c_str());
+
+    try {
+        res = name_firstupdate (jsonRequest);
+    }
+    catch (const UniValue& e) {
+        UniValue message = find_value( e, "message");
+        errorStr = message.get_str();
+        LogPrintf ("name_firstupdate error: %s\n", errorStr.c_str());
+    }
+    return errorStr;
+}
+
+void WalletModel::sendPendingNameFirstUpdates()
+{
+    for (std::map<std::string, NameNewReturn>::iterator i = pendingNameFirstUpdate.begin();
+         i != pendingNameFirstUpdate.end(); )
+    {
+        JSONRPCRequest jsonRequest;
+        UniValue params1(UniValue::VOBJ);
+        UniValue res1, val;
+        // hold the error returned from name_firstupdate, via
+        // completePendingNameFirstUpdate, or empty on success
+        // this will drive the error-handling popup
+        std::string completedResult;
+
+        std::string name = i->first;
+        std::string txid = i->second.hex;
+        std::string rand = i->second.rand;
+        std::string data = i->second.data;
+        std::string toaddress = i->second.toaddress;
+
+        params1.push_back (Pair("txid", txid));
+        jsonRequest.params = params1;
+        // if we're here, the names doesn't exist
+        // should we remove it from the DB?
+        try {
+            res1 = gettransaction( jsonRequest);
+        }
+        catch (const UniValue& e) {
+            UniValue message = find_value( e, "message");
+            std::string errorStr = message.get_str();
+            LogPrintf ("gettransaction error for name %s: %s\n",
+                       name.c_str(), errorStr.c_str());
+            ++i;
+            continue;
+        }
+
+        val = find_value (res1, "confirmations");
+        if (!val.isNum ())
+        {
+            LogPrintf ("No confirmations for name %s\n", name.c_str());
+            ++i;
+            continue;
+        }
+
+        const int confirms = val.get_int ();
+        LogPrintf ("Pending Name FirstUpdate Confirms: %d\n", confirms);
+
+        if ( confirms < 12)
```

Comment:
Please use a named constant instead of the magic number 12 here (presumably you can just reuse `MIN_FIRSTUPDATE_DEPTH`).